### ISSUE TITLE
Exporting RID generated in tool-script crashes the game without error

### ISSUE THREAD

**Issue by lilizoey** [godotengine/godot#78227]
### Godot version

4.0.3.stable

### System information

Linux 5.10.180-1-MANJARO

### Issue description

Running this code:

```gdscript
@tool
extends Node2D

@export var foo: RID = RID()

func _ready():
	foo = PhysicsServer2D.area_create()
```

Will make the game crash immediately upon running it, without providing an error message.

I understand that this is a wrong thing to do, but it seems weird to me that there is no error message in the editor, nor the terminal i opened godot from.

### Steps to reproduce

1. Create script with
```gdscript
@tool
extends Node2D

@export var foo: RID = RID()

func _ready():
	foo = PhysicsServer2D.area_create()
```
2. Add script to node in scene
3. save + reopen scene, note that the node now has an RID displayed in the inspector
4. run game
5. game immediately closes upon loading

### Minimal reproduction project

N/A

**Comment by Rindbee** [comment]
Because the script is a tool script, foo is not RID(), and this value is not fully recorded in the tscn file.

```node_2d.tscn
[gd_scene load_steps=2 format=3 uid="uid://dpuj17ur107gx"]

[ext_resource type="Script" path="res://node_2d.gd" id="1_fa10k"]

[node name="Node2D" type="Node2D"]
script = ExtResource("1_fa10k")
foo = 
```

**Comment by AThousandShips** [comment]
Fixed by: #78517
